markwon/src/main/java/ru/noties/markwon/renderer/SpannableBuilder.java

```diff
@@ -0,0 +1,63 @@
+package ru.noties.markwon.renderer;
+
+import android.support.annotation.Nullable;
+import android.text.SpannableStringBuilder;
+
+/**
+ * Copied with modification and renaming from @see <a href="https://github.com/Uncodin/bypass/blob/master/platform/android/
+ * library/src/in/uncod/android/bypass/ReverseSpannableStringBuilder.java">Uncodin/bypass</a>
+ */
+public class SpannableBuilder extends SpannableStringBuilder {
+
+    public SpannableBuilder() {
+        super();
+    }
+
+    SpannableBuilder(CharSequence text, int start, int end) {
+        super(text, start, end);
+    }
+
+    @Override
+    public <T> T[] getSpans(int queryStart, int queryEnd, Class<T> kind) {
+        T[] ret = super.getSpans(queryStart, queryEnd, kind);
+        reverse(ret);
+        return ret;
+    }
+
+    @Override
+    public void setSpan(@Nullable Object what, int start, int end, int flags) {
```

Comment:
We do not need here a recursive call. We support `Object` and `Object[]` (which can contain only `Object`s and not arrays). Maybe you have missed `super.setSpan()` on the line 32?

I also think we should introduce another method that will explicitly state what it does. Re-using `setSpan` IMO is not a very good idea. Maybe a simple `setSpans(Object spans, ...)` would be better. But anyway this kind of method must contain at least short javadoc indicating what this method is about.

Reply:
I also think that we must forbid passing null spans here. Annotate new method argument with `@NonNull`

Reply:
You are right but technically even an `Object` within an `Object[]` could potentially be an array by itself. We could simply state that nested arrays aren't supported but I just wanted to make sure one such type is not slipped trough without the array being "opened".

Reply:
Yes, I know, but we need to limit this behavior by allowing one array at maximum. To make sure that we are not allying nested array as a span, we could introduce another check inside `for` loop and throw an exception. This might be a bit of overkill, but this way we will make sure that this method is used correctly